receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -308,8 +296,25 @@ func (prw *prometheusRemoteWriteReceiver) translateV2(_ context.Context, req *wr
 		unit := req.Symbols[ts.Metadata.UnitRef]
 		description := req.Symbols[ts.Metadata.HelpRef]
 
-		resourceID := identity.OfResource(rm.Resource())
+		// Handle histograms separately due to their complex mixed-schema processing
+		if ts.Metadata.Type == writev2.Metadata_METRIC_TYPE_HISTOGRAM {
+			prw.processHistogramTimeSeries(otelMetrics, ls, ts, scopeName, scopeVersion, metricName, unit, description, metricCache, &stats)
```

Comment:
I've switched our strategy here because we can't loop over histograms and always create ExplicitHistogram or Exponential histograms for all. We need to look on a per-histogram basis, and then the implementation got a bit uglier 😕

Reply:
For consistency, I feel it would be better to do a validation step first , filter out the "bad" histograms and if there's any left , we can continue with the normal workflow.

Not sure if we're allowed to alter the incoming `req.Timeseries`, if not, you could skip any individual timeseries if any sample in it fails the validation. That would apply to classic histograms series and incompatible native histogram series (has some invalid sample).